src/main/java/net/sf/jabref/logic/citationstyle/CitationStyleToClipboardWorker.java

```diff
@@ -63,45 +64,54 @@ protected String doInBackground() throws Exception {
-
-                            "      </fo:flow>" + OS.NEWLINE +
-                            "   </fo:page-sequence>" + OS.NEWLINE +
-                            "</fo:root>" + OS.NEWLINE;
-                    break;
-                case HTML:
-                    result = "<!DOCTYPE html>" + OS.NEWLINE +
-                            "<html>" + OS.NEWLINE +
-                            "   <head>" + OS.NEWLINE +
-                            "      <meta charset=\\\"utf-8\\\">" + OS.NEWLINE +
-                            "   </head>" + OS.NEWLINE +
-                            "   <body>" + OS.NEWLINE +
-
-                            OS.NEWLINE + result + get() + OS.NEWLINE +
-
-                            "   </body>" + OS.NEWLINE +
-                            "</html>" + OS.NEWLINE;
-                    break;
+            if (CitationStyle.isCitationStyleFile(style)) {
+                String result = get();
+                switch (outputFormat) {
+                    case FO:
+                        result = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" + OS.NEWLINE +
+                                "<fo:root xmlns:fo=\"http://www.w3.org/1999/XSL/Format\">" + OS.NEWLINE +
+                                "   <fo:layout-master-set>" + OS.NEWLINE +
+                                "      <fo:simple-page-master master-name=\"citations\">" + OS.NEWLINE +
+                                "         <fo:region-body/>" + OS.NEWLINE +
+                                "      </fo:simple-page-master>" + OS.NEWLINE +
+                                "   </fo:layout-master-set>" + OS.NEWLINE +
+                                "   <fo:page-sequence master-reference=\"citations\">" + OS.NEWLINE +
+                                "      <fo:flow flow-name=\"xsl-region-body\">" + OS.NEWLINE +
+
+                                OS.NEWLINE + result + OS.NEWLINE +
+
+                                "      </fo:flow>" + OS.NEWLINE +
+                                "   </fo:page-sequence>" + OS.NEWLINE +
+                                "</fo:root>" + OS.NEWLINE;
+                        break;
+                    case HTML:
+                        result = "<!DOCTYPE html>" + OS.NEWLINE +
+                                "<html>" + OS.NEWLINE +
+                                "   <head>" + OS.NEWLINE +
+                                "      <meta charset=\\\"utf-8\\\">" + OS.NEWLINE +
+                                "   </head>" + OS.NEWLINE +
+                                "   <body>" + OS.NEWLINE +
+
+                                OS.NEWLINE + result + get() + OS.NEWLINE +
+
+                                "   </body>" + OS.NEWLINE +
+                                "</html>" + OS.NEWLINE;
+                        break;
+                }
+
+                new ClipBoardManager().setClipboardContents(result);
+
+            } else {
+                JEditorPane jEditorPane = new JEditorPane();
```

Comment:
Please add a comment that this is less an effort than using a solution of http://stackoverflow.com/q/240546/873282 to convert HTML to text and to use HtmlTransferable.